src/featureforest/utils/downloader.py

```diff
@@ -0,0 +1,45 @@
+from pathlib import Path
+import pooch
+
+
+MODELS_CACHE_DIR = Path.home().joinpath(".featureforest").joinpath("models")
+MODELS_CACHE_DIR.mkdir(parents=True, exist_ok=True)
+
+
+def is_model_exists(model_name: str) -> bool:
+    model_file = MODELS_CACHE_DIR.joinpath(model_name)
+    return model_file.exists()
+
+
+def download_model(
+    model_url: str, model_name: str,
+    cache_dir: Path = MODELS_CACHE_DIR, is_archived: bool = False
+) -> str:
+    """Download a model weights from a given url.
+
+    Args:
+        model_url (str): the model weights' url.
+        model_name (str): model's name that will be saved in cache.
+        cache_dir (Path, optional): download directory. Defaults to CACHE_DIR.
+        is_archived (bool, optional):   set to True to unzip the downloaded file.
+                                        Defaults to False.
+
+    Returns:
+        str: full path of the downloaded file.
+    """
+    if is_model_exists(model_name):
```

Comment:
technically, pooch will not download it again if it is already there, because it has a caching system.

Reply:
Try it out, but you might not need to have the if statement and pooch might directly give you `downloaded_file` without download

Reply:
Redundant codes were removed.